app/scripts/lib/multichain-api/wallet-getPermissions.test.js

```diff
@@ -66,50 +68,83 @@ const createMockedHandler = () => {
       },
     }),
   );
+  const getAccounts = jest.fn().mockResolvedValue([]);
   const response = {};
   const handler = (request) =>
     getPermissionsHandler.implementation(request, response, next, end, {
       getPermissionsForOrigin,
+      getAccounts,
     });
 
   return {
     response,
     next,
     end,
     getPermissionsForOrigin,
+    getAccounts,
     handler,
   };
 };
 
 describe('getPermissionsHandler', () => {
-  it('gets the permissions for the origin', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  beforeEach(() => {
+    MockPermittedChainsAdapters.getPermittedEthChainIds.mockReturnValue([]);
+  });
+
+  it('gets the permissions for the origin', async () => {
     const { handler, getPermissionsForOrigin } = createMockedHandler();
 
-    handler(baseRequest);
+    await handler(baseRequest);
     expect(getPermissionsForOrigin).toHaveBeenCalled();
   });
 
-  describe('BARAD_DUR flag is not set', () => {
-    beforeAll(() => {
-      delete process.env.BARAD_DUR;
-    });
-
-    it('returns `eth_accounts` restricted method typed permissions', () => {
-      const { handler, response } = createMockedHandler();
+  it('returns permissions unmodified if no CAIP-25 endowment permission has been granted', async () => {
+    const { handler, getPermissionsForOrigin, response } =
+      createMockedHandler();
 
-      handler(baseRequest);
-      expect(response.result).toStrictEqual([
-        {
+    getPermissionsForOrigin.mockReturnValue(
+      Object.freeze({
+        otherPermission: {
           id: '1',
-          parentCapability: 'eth_accounts',
+          parentCapability: 'otherPermission',
           caveats: [
             {
-              value: ['0xdead', '0xbeef'],
+              value: {
+                foo: 'bar',
+              },
             },
           ],
         },
+      }),
+    );
+
+    await handler(baseRequest);
+    expect(response.result).toStrictEqual([
+      {
+        id: '1',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    ]);
+  });
+
+  describe('CAIP-25 endowment permissions has been granted', () => {
+    it('returns the permissions with the CAIP-25 permission removed', async () => {
```

Comment:
🤔 shouldn't this be returning the permissions embedded in the CAIP-25 endowment in their legacy formats (`endowment:permittedChains` & `eth_accounts`)?

Reply:
oh I see this is an artifact of the fact that the mocks aren't actually realistic. The `getPermissionsForOrigin` mock has permissions but `getAccounts` does not

Reply:
Maybe we should keep the mocks in sync for legibility?

Reply:
Done here https://github.com/MetaMask/metamask-extension/pull/26483/commits/247c5721921dc4a747942ad2c6d18c7c46487214